src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets

```diff
@@ -88,6 +88,10 @@ Copyright (c) .NET Foundation. All rights reserved.
     <_UnsupportedTargetFrameworkError>true</_UnsupportedTargetFrameworkError>
   </PropertyGroup>
 
+  <PropertyGroup Condition=" '$(ImportWindowsDesktopTargets)' == '' And ('$(UseWpf)' == 'true' Or '$(UseWindowsForms)' == 'true')">
```

Comment:
I'm not sure TargetFrameworkInference is the best place for this check.  I would probably just put it in Microsoft.NET.Sdk.targets.